### PR TITLE
Articleselect

### PR DESCRIPTION
**Pull request by Quinhook**
Added Content:
- SupportView(Only View no funtion)
- ArticleListScreen

### CODE REVIEW

File: app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt [hunk 1]
```diff
@@ -1,13 +1,99 @@
...
+    ){
+        items(articleCategories){ category ->
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ){
+                val buttonColors = ButtonDefaults.buttonColors(Color.Green)
+               /* THIS IS FOR THE NAVIGATION LATER
```
Comment (defaultdino) [inline app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt]: Remove comments

File: app/src/main/java/android/app/lotus/view/account/SupportView.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+@Composable
+fun SupportView(){
+    var message by remember { mutableStateOf("") }
```
Comment (defaultdino) [inline app/src/main/java/android/app/lotus/view/account/SupportView.kt]: ViewModel needs to be created to handle view state, don't use local state

**Review by defaultdino — CHANGES_REQUESTED** [review]

File: app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt (hunk 1 above)
Comment (Quinhook) [reply app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt]: The code i n the comments is to be used later for when you start the navigation. Are you sure that you wnat me to take it away?

Comment (defaultdino) [reply app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt]: Yep

File: app/src/main/java/android/app/lotus/view/account/SupportView.kt [hunk 3]
```diff
@@ -22,10 +22,13 @@ import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.ViewModel
 
+class SupportViewModel : ViewModel(){
```
Comment (defaultdino) [inline app/src/main/java/android/app/lotus/view/account/SupportView.kt]: - ViewModel needs to implement @Inject constructor for Hilt/Dagger conformance
- ViewModel should be moved to Observables/ folder

File: app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt [hunk 4]
```diff
@@ -57,38 +57,22 @@ fun ScrollableList(articleCategories: List<String>, accountType: String){
...
-                    //ERROR
-                }
-
-                */
+                val buttonColors = ButtonDefaults.buttonColors(Color.Blue)
 
                 Button(
                     onClick = {
-                        //Just for visualy seeing. This is to be deleted later
+                        if(accountType == "hr"){
```
Comment (defaultdino) [inline app/src/main/java/android/app/lotus/view/home/articles/ArticleListScreen.kt]: You can leave this onClick empty - this is not how the code will look later anyways

File: app/src/main/java/android/app/lotus/view/account/SupportView.kt [hunk 5]
```diff
@@ -22,10 +22,13 @@ import androidx.compose.ui.graphics.Color
...
+    var message: String by mutableStateOf("")
+}
 @Composable
-fun SupportView(){
-    var message by remember { mutableStateOf("") }
+fun SupportView(viewModel: SupportViewModel){
```
Comment (defaultdino) [inline app/src/main/java/android/app/lotus/view/account/SupportView.kt]: ViewModel should not be passed to this view. Instantiate the viewmodel inside the composable with hiltViewModel() after fixing the above comment.

**Review by defaultdino — CHANGES_REQUESTED** [review]

**Review by defaultdino — APPROVED** [review]

**Pull request merged**